src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -36,6 +36,29 @@ public sealed class LogRecord
 
         private List<object?>? bufferedScopes;
 
+        public LogRecord(
```

Comment:
How about having a ctor which takes no parameter, and have public setters? I think this makes it easier to add additional fields in the future without having to introduce new ctors or struggle with the ordering of parameters.